src/vm/vm_core.rs

```diff
@@ -47,7 +51,7 @@ pub struct VirtualMachine {
     pub(crate) segments: MemorySegmentManager,
     pub(crate) _program_base: Option<MaybeRelocatable>,
     pub(crate) memory: Memory,
-    accessed_addresses: Option<Vec<Relocatable>>,
```

Comment:
This was made a `Vec` for a reason, `HashSet` lookups were introducing significant overhead. Just appending is enough to mark a position as accessed as relocation will take care of uniqueness later on.

Reply:
At least for now. If it proves to add too much overhead later on we can change it.